frontend/rolecall/src/app/performance/performance-editor.component.ts

```diff
@@ -152,10 +152,12 @@ export class PerformanceEditor implements OnInit, OnDestroy, AfterViewChecked {
     this.initStep3Data();
     this.initStep4();
     this.state.status = PerformanceStatus.DRAFT;
-    // Saving casts results in an immediate cast load.
-    // Clear a key parameter to prevent deadly embrace
-    this.castDnD.castSelected = false;
-
+    if (this.castDnD) {
+      // Saving casts results in an immediate cast load.
+      // Clear key parameters prevent deadly embrace.
+      this.castDnD.castSelected = false;
+      this.castDnD.selectedCastUUID = undefined;
```

Comment:
The `selectedCastUUID ` variable definition has the wrong type if it's valid to assign 'undefined' to it. If it's really valid, then the type must be changed to:
`selectedCastUUID: APITypes.CastUUID | undefined;`

Reply:
selectedCastUUID is already assigned undefined (in the delete routine and in several routines where a passed in function parameter can be undefined). An alternative would be to use and empty string as the "null value".